example/lib/mail_page.dart

```diff
@@ -0,0 +1,200 @@
+          msg: 'Hi, There is new question for you',
+          date: '22 May',
+          isUnread: false,
+        ),
+        Mail(
+          sender: 'Google',
+          sub: 'Flutter 1.5',
+          msg: 'We have launched Flutter 1.5',
+          date: '20 May',
+          isUnread: true,
+        ),
+        Mail(
+          sender: 'Github',
+          sub: 'Showcase View',
+          msg: 'New star on your showcase view.',
+          date: '21 May ',
+          isUnread: false,
+        ),
+        Mail(
+          sender: 'Simform',
+          sub: 'Credit card Plugin',
+          msg: 'Check out our credit card plugin',
+          date: '19 May',
+          isUnread: true,
+        ),
+      ],
+    );
+  }
+
+  void _getData() {
+    _addData();
+    Future.delayed(const Duration(milliseconds: 1500));
+    setState(() {
+      apiState = ApiState.success;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.white,
+      body: SafeArea(
+        bottom: false,
+        child: Column(
+          children: <Widget>[
+            const SizedBox(
+              height: 20,
+            ),
+            const Padding(padding: EdgeInsets.only(top: 8)),
+            TextButton(
+              onPressed: _getData,
+              child: const Text('Tap here to get Data'),
+            ),
+            Visibility(
+              visible: apiState != ApiState.success,
+              child: DropdownButton<String>(
+                value: dropdownValue,
+                icon: const Icon(Icons.arrow_drop_down),
+                elevation: 16,
+                style: const TextStyle(
+                  color: Colors.blue,
+                  fontWeight: FontWeight.bold,
+                ),
+                underline: Container(
+                  height: 2,
+                  color: Colors.redAccent,
+                ),
+                onChanged: (String? newValue) {
+                  _changePlaceHolderImage(newValue);
+                  setState(() {
+                    dropdownValue = newValue!;
+                  });
+                },
+                items: <String>[
+                  'Image',
+                  'SVG',
+                  'Flare',
+                  'Rive',
+                  'Rive HTTP',
+                  'GIF'
+                ].map<DropdownMenuItem<String>>((String value) {
+                  return DropdownMenuItem<String>(
+                    value: value,
+                    child: Text(value),
+                  );
+                }).toList(),
+              ),
+            ),
+            Expanded(
+              child: mails.isNotEmpty
+                  ? ListView.builder(
+                      physics: const BouncingScrollPhysics(),
+                      itemBuilder: (context, index) {
+                        return MailTile(mails[index % mails.length]);
+                      })
+                  : EmptyListPlaceholder(
```

Comment:
Let's include child parameter in the widget so the user doesn't have to explicitly put conditions to show or hide the success widget.

Reply:
Please remove the "list" from the class name as it is not only for the list.